test/HealthChecks.SendGrid.Tests/DependencyInjection/RegistrationTests.cs

```diff
@@ -19,6 +19,30 @@ public void add_health_check_when_properly_configured()
         check.ShouldBeOfType<SendGridHealthCheck>();
     }
 
+    private class SendGridOptions
+    {
+        public string ApiKey { get; set; } = default!;
+    }
+
+    [Fact]
+    public void add_health_check_from_DI_when_properly_configured()
+    {
+        var services = new ServiceCollection();
+
+        services.AddOptions<SendGridOptions>().Configure(options => options.ApiKey = "my_api_key");
+        services.AddHealthChecks()
+            .AddSendGrid(sp => sp.GetRequiredService<IOptions<SendGridOptions>>().Value.ApiKey);
+
+        using var serviceProvider = services.BuildServiceProvider();
+        var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+        var registration = options.Value.Registrations.First();
+        var check = registration.Factory(serviceProvider);
+
+        registration.Name.ShouldBe("sendgrid");
+        check.ShouldBeOfType<SendGridHealthCheck>();
```

Comment:
It would be nice to at least check that given overload was used.

```suggestion
        bool called = false;
        var services = new ServiceCollection();

        services.AddOptions<SendGridOptions>().Configure(options => options.ApiKey = "my_api_key");
        services.AddHealthChecks()
            .AddSendGrid(sp =>
            {
                called = true;
                return sp.GetRequiredService<IOptions<SendGridOptions>>().Value.ApiKey;
            });

        using var serviceProvider = services.BuildServiceProvider();
        var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();

        var registration = options.Value.Registrations.First();
        var check = registration.Factory(serviceProvider);

        registration.Name.ShouldBe("sendgrid");
        check.ShouldBeOfType<SendGridHealthCheck>();
        called.ShouldBeTrue();
```